examples/isometric/game.go

```diff
@@ -0,0 +1,217 @@
+		g.camScale -= (g.camScale - g.camScaleTo) / div
+	}
+
+	pan := 4.0 / g.camScale
+	// Pan camera via keyboard.
+	if ebiten.IsKeyPressed(ebiten.KeyLeft) || ebiten.IsKeyPressed(ebiten.KeyA) {
+		g.camX -= pan
+	}
+	if ebiten.IsKeyPressed(ebiten.KeyRight) || ebiten.IsKeyPressed(ebiten.KeyD) {
+		g.camX += pan
+	}
+	if ebiten.IsKeyPressed(ebiten.KeyDown) || ebiten.IsKeyPressed(ebiten.KeyS) {
+		g.camY -= pan
+	}
+	if ebiten.IsKeyPressed(ebiten.KeyUp) || ebiten.IsKeyPressed(ebiten.KeyW) {
+		g.camY += pan
+	}
+	// Pan camera via mouse.
+	if ebiten.IsMouseButtonPressed(ebiten.MouseButtonRight) {
+		if g.mousePanX == 0 && g.mousePanY == 0 {
+			g.mousePanX, g.mousePanY = ebiten.CursorPosition()
+		} else {
+			x, y := ebiten.CursorPosition()
+			dx, dy := float64(g.mousePanX-x)*(pan/100), float64(g.mousePanY-y)*(pan/100)
+			g.camX, g.camY = g.camX-dx, g.camY+dy
+		}
+	} else if g.mousePanX != 0 || g.mousePanY != 0 {
+		g.mousePanX, g.mousePanY = 0, 0
+	}
+
+	// Clamp camera position.
+	levelWidth := len(g.currentLevel.tiles)
+	levelHeight := len(g.currentLevel.tiles[0])
+	worldWidth := float64(levelWidth * g.currentLevel.tileSize / 2)
+	worldHeight := float64(levelHeight * g.currentLevel.tileSize / 2)
+	if g.camX < worldWidth*-1 {
+		g.camX = worldWidth * -1
+	} else if g.camX > worldWidth {
+		g.camX = worldWidth
+	}
+	if g.camY < worldHeight*-1 {
+		g.camY = worldHeight * -1
+	} else if g.camY > 0 {
+		g.camY = 0
+	}
+
+	// Randomize level.
+	if inpututil.IsKeyJustPressed(ebiten.KeyR) {
+		g.currentLevel = NewLevel()
+	}
+
+	return nil
+}
+
+// Draw draws the Game on the screen.
+func (g *Game) Draw(screen *ebiten.Image) {
+	// Render level.
+	g.renderLevel(screen)
+
+	// Print game info.
+	debugBox := image.NewRGBA(image.Rect(0, 0, g.w, 200))
+	debugImg := ebiten.NewImageFromImage(debugBox)
+	ebitenutil.DebugPrint(debugImg, fmt.Sprintf("KEYS WASD EC R\nFPS  %0.0f %c\nTPS  %0.0f\nSCA  %0.3f\nPOS  %0.1f,%0.1f", ebiten.CurrentFPS(), spinner[g.spinnerIndex], ebiten.CurrentTPS(), g.camScale, g.camX, g.camY))
+	op := &ebiten.DrawImageOptions{Filter: ebiten.FilterNearest}
+	op.GeoM.Translate(3, 0)
+	op.GeoM.Scale(2, 2)
+	screen.DrawImage(debugImg, op)
+
+	// Advance FPS spinner.
+	g.spinnerIndex++
+	if g.spinnerIndex == 4 {
+		g.spinnerIndex = 0
+	}
+}
+
+// Layout is called when the Game's layout changes.
+func (g *Game) Layout(outsideWidth, outsideHeight int) (int, int) {
+	s := ebiten.DeviceScaleFactor()
+	g.w, g.h = int(s*float64(outsideWidth)), int(s*float64(outsideHeight))
+	return g.w, g.h
+}
+
+// cartesianToIso transforms cartesian coordinates into isometric coordinates.
+func (g *Game) cartesianToIso(x, y float64) (float64, float64) {
+	tileSize := g.currentLevel.tileSize
+	ix := (x - y) * float64(tileSize/2)
+	iy := (x + y) * float64(tileSize/4)
+	return ix, iy
+}
+
+// isoToCartesian transforms isometric coordinates into cartesian coordinates.
+func (g *Game) isoToCartesian(x, y float64) (float64, float64) {
```

Comment:
This function is not used anywhere.

Reply:
As an example, I don't want to have a function that is not used anywhere.

Reply:
Including this function makes the example more complete IMO as it is necessary to support interaction (namely mouse).  Anyone who wants to actually make something using this demo as a starting point will need it.  Nevertheless, I've removed it.

Reply:
I understand this function might be useful for users. If you want to leave this, I'm fine to leave this as a comment here.

Reply:
Are you asking if I'd like to document this function by adding it to a PR review comment?  Sorry if I am misunderstanding.

If that is what you meant, I believe the function is already documented in this way as it is attached to this comment thread above.  Thank you ,though.

Reply:
Derp, I just realized what you meant. Including this function within a comment block sounds great.  I've updated the PR.

Reply:
I think more comments are needed to explain, like, "this function might be useful for those who want to extend this example".